src/coreclr/vm/codeman.cpp

```diff
@@ -4605,118 +4621,140 @@ RangeSection* ExecutionManager::GetRangeSection(TADDR addr)
         SUPPORTS_DAC;
     } CONTRACTL_END;
 
-    RangeSection * pHead = m_CodeRangeList;
-
-    if (pHead == NULL)
+    if (m_RangeSectionHandleReaderHeader == nullptr)
     {
         return NULL;
     }
 
-    RangeSection *pCurr = pHead;
-    RangeSection *pLast = NULL;
-
+    ReaderLockHolder rlh;
```

Comment:
We need whatever lock we are using to protect not only the rangesection lists, but also the RangeSection instances themselves. This is not an appropriate place to take the lock.